coalib/bearlib/__init__.py

```diff
@@ -38,6 +38,20 @@ def deprecate_settings(**depr_args):
      >>> func(new='coala!')
      coala!
 
+     This example represents the case where the old and new settings are
+     provided to the function.
+
+     >>> @deprecate_settings(new='old')
+     ... def run(new):
+     ...     print(new)
+     >>> run(old="Hello!", new='coala is always written with lowercase `c`.')
+     The setting `old` is deprecated. Please use `new` instead.
+     The value of `old` and `new` are conflicting. `new` will be used instead.
+     coala is always written with lowercase `c`.
```

Comment:
what's this line do? I suggest doing at least a very short self review before submitting something, I usually try to do that.